File: Documentation/4.0/Samples/python/clientapi/session/StoringEntities.py
from pyravendb.store.document_store import DocumentStore


class Employee(object):
    def __init__(self, first_name, last_name):
        self.first_name = first_name
        self.last_name = last_name


class StoringEntities(object):
    @staticmethod
    def open_session():
        store = DocumentStore()
        store.initialize()

        # region store_entities_1
        def store(self, entity, key=None, change_vector=None):
            # endregion
            pass

        with store.open_session() as session:
            entity = Employee("first_name", "second_name")
            change_vector = ""
            # region store_entities_2
            session.store(entity)
            # endregion

            # region store_entities_3
            session.store(entity, change_vector=change_vector)
            # endregion

            # region store_entities_4
            session.store(entity, key="doc/1")
            # endregion

            # region store_entities_5
            session.store(entity, key="doc/1", change_vector=change_vector)
            # endregion

        with store.open_session() as session:
            # region store_entities_6
            # generate Id automatically
            # when we have a new and empty database and conventions are not changed: 'employees/1-A'
            employee = Employee("John", "Doe")
            session.store(employee)
            session.save_changes()
            # endregion

File: Documentation/4.0/Samples/nodejs/client-api/session/storingEntities.js
import { DocumentStore } from "ravendb";

const store = new DocumentStore();
const session = store.openSession();

//region store_entities_1
session.store(entity); 
//endregion

//region store_entities_2
session.store(entity, id);
//endregion

//region store_entities_3
session.store(entity, id, options);
//endregion

class Employee {
    constructor(firstName, lastName) {
        this.firstName = firstName;
        this.lastName = lastName;
    }
}

const session = store.openSession();
//region store_entities_5
const employee = new Employee("John", "Doe");

// generate Id automatically
await session.store(employee);

// send all pending operations to server, in this case only `Put` operation
await session.saveChanges();
//endregion

//region storing_literals_1
const store = new DocumentStore(urls, database);
store.conventions.findCollectionNameForObjectLiteral = entity => entity["collection"];
// ...
store.initialize();
//endregion
